src/models/version.rs

```diff
@@ -172,6 +177,10 @@ impl NewVersion {
                 )));
             }
 
+            if let Some(limit) = daily_limit {
+                self.rate_limit(conn, limit)?;
+            }
```

Comment:
I assume you've put the rate limiting here to ensure that it's running in the same database transaction? Since we have the `publish_rate_limit` already in the publish controller, I think that might be a more straight forward place for it. That way we also don't need to adjust all callers of the `save()` method to add this additional parameter.

Reply:
I was indeed trying to separate the concepts, since I saw the existing rate limiter uses a bucket system, and here I'm simply querying on a time range.
I can move the method inside the PublishRateLimit struct without problems

Reply:
> I can move the method inside the PublishRateLimit struct without problems

not necessarily inside the struct, but I think it makes more sense to do the time range query in the controller, instead of in the model method :)